proto/substrait/algebra.proto

```diff
@@ -267,11 +267,23 @@ message CrossRel {
 message FetchRel {
   RelCommon common = 1;
   Rel input = 2;
-  // the offset expressed in number of records
-  int64 offset = 3;
-  // the amount of records to return
-  // use -1 to signal that ALL records should be returned
-  int64 count = 4;
+  oneof offset_mode {
+    // the offset expressed in number of records
+    // Deprecated: use `offset_expr` instead
+    int64 offset = 3 [deprecated = true];
+    // Expression evaluated into an integer specifying the number of records to
+    // skip.
+    Expression offset_expr = 5;
+  }
```

Comment:
I may not know protobuf well enough: is this a recommended way to evolve a message type? It seems like `FetchRel` does not have an `offset` field anymore -- isn't that more breakage than necessary (compared with *only* marking that field `decprecated`)?

Reply:
Through `oneof`, `offset` remains exposed as a top-level field, but it prevents both fields from being set simultaneously. Initially, I used the approach you proposed, but later switched to this method based on @jacques-n's recommendation in the mailing list discussion, which also appears to align with the process followed in similar changes, e.g., #665.